### PR TITLE
Some changes to the wallclock node

### PR DESCRIPTION
**Pull request by cabelitos**
@barbieri

### CODE REVIEW

File: src/modules/flow/wallclock/wallclock.c [hunk 1]
```diff
@@ -573,9 +676,58 @@ static void
...
+    r = sol_flow_packet_get_boolean(packet, &enabled);
+    SOL_INT_CHECK(r, < 0, r);
+
+    if (enabled && !mdata->registered) {
+        struct wallclock_timer *timer = &timers[mdata->type];
+
+        r = wallclock_add_client(node, mdata);
+        SOL_INT_CHECK(r, < 0, r);
+        return sol_flow_send_irange_packet(node, 0, &timer->val);
+    }
```
Comment (barbieri) [inline src/modules/flow/wallclock/wallclock.c]: check if it was registered before we try to remove the client? Similar logic you did for timeblock.

**Comment by barbieri** [comment]
overall looks good, just the minor issue trying to remove a client that is already removed, it shouldn't crash, but would do an useless loop.

**Comment by cabelitos** [comment]
@barbieri I will add:
```C
if (!enabled && mdata->registered)
....
```
Before trying to remove it.

**Comment by barbieri** [comment]
ok

**Comment by cabelitos** [comment]
merged

**Pull request closed without merging**